flocker/docs/bootstrap/_simple.py

```diff
@@ -0,0 +1,116 @@
+
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Simple directives that just wrap content.
+"""
+
+from textwrap import dedent
+
+from docutils import nodes
+from docutils.parsers.rst import Directive
+
+
+def create_simple_html_directive(name, pre, post, has_content=True, match_titles=False):
+
+    node_class = type(name.replace('-', '_'), (nodes.General, nodes.Element), {})
+
+    def visit_html(self, node):
+        self.body.append(pre)
+
+    def depart_html(self, node):
+        self.body.append(post)
+
+    def run_directive(self):
+        node = node_class()
+        if has_content:
+            text = self.content
+            self.state.nested_parse(text, self.content_offset, node, match_titles=match_titles)
+        # FIXME: This should add more stuff.
+        self.state.document.settings.record_dependencies.add(__file__)
+        return [node]
+
+    directive_class = type(name.title() + 'Directive', (Directive,), {
+        "has_content": has_content,
+        "run": run_directive,
+    })
+
+    def setup(app):
+        app.add_node(node_class,
+                     html=(visit_html, depart_html))
+        app.add_directive(name, directive_class)
+
+    return node_class, directive_class, setup
+
+
+intro_text, IntroTextDirective, intro_text_setup = (
+    create_simple_html_directive(
+        "intro-text",
+        pre=dedent("""\
+        <header><h1 class="text-center">
+        """),
+        post=dedent("""\
+        </h1></header>
+        """),
+    ))
+
+
+tutorial_step, TutorialStepDirective, tutorial_step_setup = (
+    create_simple_html_directive(
+        "tutorial-step",
+        pre=dedent("""\
+        <div class="container"><div class="row"><div class="col-md-12">
+        """),
+        post=dedent("""\
+        </div></div></div>
+        """),
+        match_titles=True,
+    ))
+
+
+mobile_label, MobileLabelDirective, mobile_label_setup = (
```

Comment:
This probably wants to change to use an argument, rather than content, but this is good enough for a first pass.